File: app/src/main/java/kr/hs/emirim/sookhee/redonorpets/MainActivity.java
package kr.hs.emirim.sookhee.redonorpets;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.Toast;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    private FragmentManager fragmentManager = getSupportFragmentManager();
    private FragmentShelter fragmentShelter = new FragmentShelter();
    private FragmentHome fragmentHome = new FragmentHome();
    private FragmentMypage fragmentMypage = new FragmentMypage();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SharedPreferences pref = getSharedPreferences("pref", MODE_PRIVATE);
        boolean isLogin = pref.getBoolean("isLogin", false);

        if(isLogin == false) {
            Intent intent = new Intent(getApplicationContext(), LoginActivity.class);
            startActivity(intent);
            finish();
        }
        else{
            super.onCreate(savedInstanceState);
            setContentView(R.layout.activity_main);

            FragmentTransaction transaction = fragmentManager.beginTransaction();
            transaction.replace(R.id.frameLayout, fragmentHome).commitAllowingStateLoss();

            BottomNavigationView bottomNavigationView = findViewById(R.id.navigationView);
            bottomNavigationView.setOnNavigationItemSelectedListener(new ItemSelectedListener());
            bottomNavigationView.setSelectedItemId(R.id.homeItem);
        }
    }

    class ItemSelectedListener implements BottomNavigationView.OnNavigationItemSelectedListener{
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
            FragmentTransaction transaction = fragmentManager.beginTransaction();

            switch(menuItem.getItemId())
            {
                case R.id.shelterItem:
                    transaction.replace(R.id.frameLayout, fragmentShelter).commitAllowingStateLoss();

                    break;
                case R.id.homeItem:
                    transaction.replace(R.id.frameLayout, fragmentHome).commitAllowingStateLoss();
                    break;
                case R.id.mypageItem:
                    transaction.replace(R.id.frameLayout, fragmentMypage).commitAllowingStateLoss();
                    break;
            }
            return true;
        }
    }
}

File: app/src/main/java/kr/hs/emirim/sookhee/redonorpets/ui/main/MainActivity.kt
package kr.hs.emirim.sookhee.redonorpets.ui.main

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import kr.hs.emirim.sookhee.redonorpets.*
import kr.hs.emirim.sookhee.redonorpets.databinding.ActivityMainBinding
import kr.hs.emirim.sookhee.redonorpets.ui.home.HomeFragment

/**
 *  MainActivity.kt
 *
 *  Created by Minji Jeong on 2022/01/06
 *  Copyright © 2022 DonorPets2.0. All rights reserved.
 */

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    private val shelterFragment by lazy { FragmentShelter() }
    private val homeFragment by lazy { HomeFragment() }
    private val mypageFragment by lazy { FragmentMypage() }
    private var activeFragment = homeFragment as Fragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding.root)

        checkIsLogin()
    }

    private fun checkIsLogin() {
        val isLogin = true // TODO : 파베 연결하기 전 임시 플래그

        if (isLogin) {
            initBottomNavigation()
        } else {
            startActivity(Intent(this, LoginActivity::class.java))
        }
    }

    private fun initBottomNavigation() {
        binding.navigationView.apply {
            selectedItemId = R.id.homeItem
            setOnNavigationItemSelectedListener { menu ->
                when (menu.itemId) {
                    R.id.shelterItem -> setFragment(shelterFragment)
                    R.id.homeItem -> setFragment(homeFragment)
                    R.id.mypageItem -> setFragment(mypageFragment)
                }

                return@setOnNavigationItemSelectedListener true
            }
            setFragment(homeFragment)
        }
    }

    private fun setFragment(currentFragment: Fragment) {
        val fragmentManager = supportFragmentManager.beginTransaction()
            .setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE).hide(activeFragment)

        if (!currentFragment.isAdded) {
            fragmentManager.add(R.id.frameLayout, currentFragment, currentFragment.toString())
                .show(currentFragment).commit()
        } else if (activeFragment != currentFragment) {
            fragmentManager.show(currentFragment).commit()
        }

        activeFragment = currentFragment
    }
}
